### PR TITLE
Fixes #22689: silent rudder-upgrade failure / plugins were not upgraded when upgrading from 7.2.5 to 7.3.1~nightly

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/22689

Replacing previous PR: https://github.com/Normation/rudder-packages/pull/2722

### CODE REVIEW

File: rudder-server/SOURCES/rudder-server-postinst [hunk 1]
```diff
@@ -37,13 +37,23 @@ echo "`date` - Starting rudder-server post installation script" >> ${LOG_FILE}
...
+      echo "INFO: External database detected, skipping database creation"
+      CREATE_DB="false"
+    else
+      USER=$(grep rudder.jdbc.username /opt/rudder/etc/rudder-web.properties)
+      CHK_PG_USER=$(su postgres -c "psql -t -c \"select count(1) from pg_user where usename = '${USERNAME}'\"")
+      if [ ${CHK_PG_DB} -ne 0 ]
+      then
+        echo "INFO: Existing database detected, skipping database creation"
+        CREATE_DB="false"
+      fi
```
Comment (ncharles) [inline rudder-server/SOURCES/rudder-server-postinst:56]: you are msising a fi

**Comment by peckpeck** [comment]
PR updated with a new commit

**Review by amousset — APPROVED** [review]

File: rudder-server/SOURCES/rudder-server-postinst [hunk 2]
```diff
@@ -37,13 +37,24 @@ echo "`date` - Starting rudder-server post installation script" >> ${LOG_FILE}
...
-    echo "INFO: External database detected, skipping rudder-reports postinst"
-    CREATE_DB="false"
+    # if it is an external DB, do not run
+    if ! grep -qE 'rudder.jdbc.url=jdbc:postgresql://(localhost|127\.0\.)' /opt/rudder/etc/rudder-web.properties
+    then
+      echo "INFO: External database detected, skipping database creation"
+      CREATE_DB="false"
+    else
+      USER=$(grep rudder.jdbc.username /opt/rudder/etc/rudder-web.properties)
+      CHK_PG_USER=$(su postgres -c "psql -t -c \"select count(1) from pg_user where usename = '${USERNAME}'\"")
```
Comment (VinceMacBuche) [inline rudder-server/SOURCES/rudder-server-postinst:51]: USERNAME variable  is not created, it is USER. And the value of USER variable is wrong the command gives 

grep rudder.jdbc.username /opt/rudder/etc/rudder-web.properties 
rudder.jdbc.username=rudder

on my rudder server

Comment (ncharles) [reply rudder-server/SOURCES/rudder-server-postinst:51]: postgresql needs to be started

**Pull request merged**